src/main/java/twilightforest/entity/boss/EntityTFNaga.java

```diff
@@ -650,16 +659,13 @@ private boolean isEntityWithinHomeArea(Entity entity)
 
 	private void spawnBodySegments()
 	{
-		if (!world.isRemote)
+		for (int i = 0; i < currentSegmentCount; i++)
 		{
-			for (int i = 0; i < currentSegmentCount; i++)
+			if (body[i] == null || body[i].isDead)
 			{
-				if (body[i] == null || body[i].isDead)
-				{
-					body[i] = new EntityTFNagaSegment(this, i);
-					body[i].setLocationAndAngles(posX + 0.1 * i, posY + 0.5D, posZ + 0.1 * i, rand.nextFloat() * 360F, 0.0F);
-					world.spawnEntity(body[i]);
-				}
+				body[i] = new EntityTFNagaSegment(this, i);
+				body[i].setLocationAndAngles(posX + 0.1 * i, posY + 0.5D, posZ + 0.1 * i, rand.nextFloat() * 360F, 0.0F);
+				if (!world.isRemote) world.spawnEntity(body[i]); // Moved the !world.isRemote check here as the segments were not spawning on the client side otherwise for some reason.
```

Comment:
yea the multipart entity stuff is confusing. It's fine for now, can clean it up later